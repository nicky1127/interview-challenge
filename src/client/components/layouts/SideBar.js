import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

const SideBar = props => {
  const { items } = props;
  return (
    <div>
      <div className="filters">
        <input className="form-control" placeholder="Name" />
      </div>
      <ul className="item-picker">
        <li className="item">
          <h2>Dummy item</h2>
          <p>
            <span className="dietary">ve</span>
            <span className="dietary">v</span>
            <span className="dietary">n!</span>
          </p>
        </li>
        <li className="item">
          <h2>Dummy item</h2>
          <p>
            <span className="dietary">ve</span>
            <span className="dietary">v</span>
            <span className="dietary">n!</span>
          </p>
        </li>
        <li className="item">
          <h2>Dummy item</h2>
          <p>
            <span className="dietary">ve</span>
            <span className="dietary">v</span>
            <span className="dietary">n!</span>
          </p>
        </li>
        <li className="item">
          <h2>Dummy item</h2>
          <p>
            <span className="dietary">ve</span>
            <span className="dietary">v</span>
            <span className="dietary">n!</span>
          </p>
        </li>
        <li className="item">
          <h2>Dummy item</h2>
          <p>
            <span className="dietary">ve</span>
            <span className="dietary">v</span>
            <span className="dietary">n!</span>
          </p>
        </li>
        <li className="item">
          <h2>Dummy item</h2>
          <p>
            <span className="dietary">ve</span>
            <span className="dietary">v</span>
            <span className="dietary">n!</span>
          </p>
        </li>
        <li className="item">
          <h2>Dummy item</h2>
          <p>
            <span className="dietary">ve</span>
            <span className="dietary">v</span>
            <span className="dietary">n!</span>
          </p>
        </li>
      </ul>
    </div>
  );
};

const mapStateToProps = state => {
  const { items } = state;
  return {
    items
  };
};

const ConnectedSideBar = connect(mapStateToProps)(SideBar);

export default ConnectedSideBar;
