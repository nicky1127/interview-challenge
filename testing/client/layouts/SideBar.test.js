import { render, cleanup} from "@testing-library/react";
import SideBar from "client/components/layouts/SideBar";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const state = {
  items: [{ name: "test", id: 1, dietaries: ["test"] }],
  ldgItems: false,
  ldgSuccItems: false
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("SideBar", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SideBar />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
    it("SideBar should render one [data-testid=sideBar]", () => {
      const { getByTestId } = render(
        <Provider store={store}>
          <SideBar />
        </Provider>
      );
      const container = getByTestId("sideBar");
      expect(container).toBeInTheDocument();
    });
});
