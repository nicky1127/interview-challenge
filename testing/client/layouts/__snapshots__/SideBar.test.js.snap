// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Snapshot matchs 1`] = `
<div
  data-testid="sideBar"
>
  <div
    className="filters"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Name"
    />
  </div>
  <ul
    className="item-picker"
  >
    <li
      className="item"
    >
      <h2>
        test
      </h2>
      <p>
        <span
          className="dietary"
          data-testid="sideBarItem"
        >
          test
        </span>
      </p>
    </li>
  </ul>
</div>
`;
